src/api/APIModel.ts

```diff
@@ -17,7 +19,9 @@ export abstract class APIModel {
 	abstract getById(id: string): Promise<MediaTypeModel>;
 
 	hasType(type: MediaType): boolean {
-		return this.types.contains(type);
+		if (this.types.contains(type) && !(this.plugin.settings[[this.apiName, type].filter(s => s).join('') as keyof typeof this.plugin.settings] === false)){
```

Comment:
`[this.apiName, type].filter(s => s).join('') as keyof typeof this.plugin.settings`

Why not do `(this.apiName + type) as keyof MediaDbPluginSettings`?

However, I would still recommend organizing the settings into sub-objects for each api.